### PR TITLE
Add test for commit message format

### PR DESCRIPTION
**Pull request by lauralt**
Fixes: #9, #14.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -190,3 +190,17 @@ steps:
...
+    commands:
+      - pytest rust-vmm-ci/integration_tests/test_commit_format.py
+    retry:
+      automatic: false
+    agents:
+      platform: x86_64.metal
+      os: linux
+    plugins:
+      - docker#v3.0.1:
+          privileged: true
```
Comment (aghecenco) [inline .buildkite/pipeline.yml:204]: Does the container need root privileges for this test?

File: integration_tests/test_commit_format.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+def test_commit_format():
+    """
+    Checks commit message format for the current PR's commits.
+
+    Checks if commit messages follow the 50/72 git commit rule
+    and if commits are signed.
+    """
+    # Get hashes of PR's commits in their abbreviated form for
+    # a prettier printing.
+    shas_cmd = "git log --no-merges --pretty=%h --no-decorate " \
```
Comment (aghecenco) [inline integration_tests/test_commit_format.py:23]: `git` won't be available in the container until v6 is published

File: integration_tests/test_commit_format.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+COMMIT_TITLE_MAX_LEN = 50
+COMMIT_BODY_LINE_MAX_LEN = 72
+BASE_BRANCH = os.environ['BUILDKITE_PULL_REQUEST_BASE_BRANCH']
+
+
+def test_commit_format():
+    """
+    Checks commit message format for the current PR's commits.
+
+    Checks if commit messages follow the 50/72 git commit rule
```
Comment (aghecenco) [inline integration_tests/test_commit_format.py:18]: nit: after the explanation, add a link to the 50/72 rule so it doesn't look like they're just arbitrary numbers we happen to like

File: integration_tests/utils.py [hunk 4]
```diff
@@ -9,3 +11,12 @@ def get_repo_root_path():
     rust_vmm_ci_path = os.path.dirname(integration_tests_path)
 
     return os.path.dirname(rust_vmm_ci_path)
+
+
+def get_cmd_output(cmd):
+    """Returns stdout content of `cmd` command."""
+    cmd_out = subprocess.run(cmd, shell=True, check=True,
+                             stdout=subprocess.PIPE,
+                             stderr=subprocess.PIPE)
```
Comment (aghecenco) [inline integration_tests/utils.py:20]: This function doesn't use `stderr` for anything, so no need to pipe it

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (lauralt) [reply .buildkite/pipeline.yml:204]: Nope, it shouldn't.

File: integration_tests/test_commit_format.py (hunk 3 above)
Comment (lauralt) [reply integration_tests/test_commit_format.py:18]: Wanted to, then I forgot :(, thanks.

File: integration_tests/utils.py (hunk 4 above)
Comment (lauralt) [reply integration_tests/utils.py:20]: Done.

File: integration_tests/test_commit_format.py (hunk 2 above)
Comment (lauralt) [reply integration_tests/test_commit_format.py:23]: Sorry, I was under the impression it was updated in the last PR from there.

Comment (aghecenco) [reply integration_tests/test_commit_format.py:24]: The container image still has to be built and pushed to the Docker registry (sounds like something we could automate :thinking: but later); in the meantime I did that and v6 should be available now

File: .buildkite/pipeline.yml [hunk 5]
```diff
@@ -188,5 +188,17 @@ steps:
...
+  - label: "commit-format"
+    commands:
+      - pytest rust-vmm-ci/integration_tests/test_commit_format.py
+    retry:
+      automatic: false
+    agents:
+      os: linux
+    plugins:
+      - docker#v3.0.1:
+          image: "rustvmm/dev:v6"
```
[the remark this answers is not in the document]
Comment (andreeaflorescu) [reply .buildkite/pipeline.yml:203]: https://github.com/buildkite-plugins/docker-buildkite-plugin

**Comment by lauralt** [comment]
It seems the dummy PRs from kvm-ioctls: https://github.com/rust-vmm/kvm-ioctls/pull/105 and https://github.com/rust-vmm/kvm-ioctls/pull/107 are failing right.

**Pull request merged**